src/main/kotlin/io/emeraldpay/dshackle/rpc/NativeCall.kt

```diff
@@ -101,19 +104,60 @@ open class NativeCall(
     }
 
     open fun nativeCallResult(requestMono: Mono<BlockchainOuterClass.NativeCallRequest>): Flux<CallResult> {
+        val requestSpan = tracer.currentSpan()
         return requestMono.flatMapMany(this::prepareCall)
             .flatMap {
-                if (it.isValid()) {
-                    val parsed = parseParams(it.get())
-                    this.fetch(parsed)
-                        .doOnError { e -> log.warn("Error during native call: ${e.message}") }
+                val requestId = it.requestId
+                val requestCount = it.requestCount
+                val id = it.getContextId()
+                val result = if (it.isValid()) {
+                    run {
+                        val parsed = try {
+                            parseParams(it.get())
+                        } catch (e: Exception) {
+                            return@run Mono.error(e)
+                        }
+                        if (requestCount == 1 && requestId.isNotBlank()) {
```

Comment:
and if request count != 1, what then?

Reply:
We will create a new span per request and pass it in reactor context